frame/im-online/src/lib.rs

```diff
@@ -408,24 +340,11 @@ pub mod pallet {
 	pub(crate) type Keys<T: Config> =
 		StorageValue<_, WeakBoundedVec<T::AuthorityId, T::MaxKeys>, ValueQuery>;
 
-	/// For each session index, we keep a mapping of `SessionIndex` and `AuthIndex` to
-	/// `WrapperOpaque<BoundedOpaqueNetworkState>`.
+	/// For each session index, we keep a mapping of `SessionIndex` and `AuthIndex`.
 	#[pallet::storage]
 	#[pallet::getter(fn received_heartbeats)]
-	pub(crate) type ReceivedHeartbeats<T: Config> = StorageDoubleMap<
-		_,
-		Twox64Concat,
-		SessionIndex,
-		Twox64Concat,
-		AuthIndex,
-		WrapperOpaque<
-			BoundedOpaqueNetworkState<
-				T::MaxPeerDataEncodingSize,
-				T::MaxPeerDataEncodingSize,
-				T::MaxPeerInHeartbeats,
-			>,
-		>,
-	>;
+	pub(crate) type ReceivedHeartbeats<T: Config> =
+		StorageDoubleMap<_, Twox64Concat, SessionIndex, Twox64Concat, AuthIndex, bool>;
```

Comment:
```suggestion
		StorageDoubleMap<_, Twox64Concat, SessionIndex, Twox64Concat, AuthIndex, ()>;
```
Why a bool when only `true` is inserted?

Reply:
This ^^

Reply:
Do we even support this? @cheme could we store empty values in the trie?

Reply:
Thought so too. Changed in https://github.com/paritytech/substrate/pull/14251/commits/dcc7b38fa3e53876a0e1d5c76b6c3ef64c157c76

Reply:
Did you ensured this works @melekes?

Reply:
I thought (maybe naively) that 500 lines of tests should cover that, no?

Reply:
Actually it does not work. I'm going to revert https://github.com/paritytech/substrate/commit/dcc7b38fa3e53876a0e1d5c76b6c3ef64c157c76